src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```
@@ -456,3 +456,11 @@ services:
     autowire: true
     arguments:
       - '@prestashop.core.query.doctrine_search_criteria_applicator'
+
+  prestashop.core.grid.query_builder.customer_groups:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\CustomerGroupsQueryBuilder'
+    parent: 'prestashop.core.grid.abstract_query_builder'
+    arguments:
+      - '@prestashop.core.query.doctrine_search_criteria_applicator'
+      - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
+    public: true
```

We had the bad habit to make all our services public so far, it's not a good habit because a service should be private by default in order to favor DI instead of manual retrieval by the container
Some of our services do need to be public but it should be the exception, so by default you should set the services as private

Since there's a generic public rule on the file (like most our definition files unfortunately) you may need to explicitly set this as private
```suggestion
    public: false
```